ee/secureenclaverunner/secureenclaverunner.go

```diff
@@ -0,0 +1,282 @@
+//go:build darwin
+// +build darwin
+
+package secureenclaverunner
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"os/user"
+	"sync"
+	"time"
+
+	"github.com/kolide/launcher/ee/agent/types"
+	"github.com/kolide/launcher/ee/consoleuser"
+	"github.com/kolide/launcher/pkg/traces"
+)
+
+const (
+	publicEccDataKey = "publicEccData"
+)
+
+type secureEnclaveRunner struct {
+	uidPubKeyMap        map[string]*ecdsa.PublicKey
+	secureEnclaveClient secureEnclaveClient
+	store               types.GetterSetterDeleter
+	slogger             *slog.Logger
+	mux                 *sync.Mutex
+	interrupt           chan struct{}
+	interrupted         bool
+}
+
+type secureEnclaveClient interface {
+	CreateSecureEnclaveKey(uid string) (*ecdsa.PublicKey, error)
+}
+
+func New(ctx context.Context, slogger *slog.Logger, store types.GetterSetterDeleter, secureEnclaveClient secureEnclaveClient) (*secureEnclaveRunner, error) {
+	ctx, span := traces.StartSpan(ctx)
+	defer span.End()
+
+	ser := &secureEnclaveRunner{
+		uidPubKeyMap:        make(map[string]*ecdsa.PublicKey),
+		store:               store,
+		secureEnclaveClient: secureEnclaveClient,
+		slogger:             slogger.With("component", "secureenclaverunner"),
+		mux:                 &sync.Mutex{},
+		interrupt:           make(chan struct{}),
+	}
+
+	data, err := store.Get([]byte(publicEccDataKey))
```

Comment:
Since this was just reading from bbolt and unmarshalling the json (without actual secure enclave op) I though it was reasonable to have it in new, but I suppose putting it in the top of execute would save a bit of start up time.

Reply:
moved to top of execute